pkg/proxy/iptables/proxier.go

```diff
@@ -986,8 +986,25 @@ func (proxier *Proxier) syncProxyRules() {
 					writeLine(proxier.natRules, append(args, "-j", string(KubeMarkDropChain))...)
 				}
 			}
+		} else {
+			// REJECT rules if there is no service endpoint
+			for _, ingress := range svcInfo.LoadBalancerStatus.Ingress {
```

Comment:
We should probably move the hasEndpoints into the inner loop to make it more readable, since we do this same iteration in both.

Reply:
Good point. I will update the PR.